packages/pytest-simcore/src/pytest_simcore/httpx_calls_capture.py

```diff
@@ -0,0 +1,273 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=too-many-arguments
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+"""
+The pytest_simcore.httpx_calls_capture module provides fixtures to capture the calls made by instances of httpx.AsyncClient
+when interacting with a real backend. These captures can then be used to create a respx.MockRouter, which emulates the backend while running
+your tests.
+
+This module ensures a reliable reproduction and maintenance of mock responses that reflect the real backend environments used for testing.
+
+## Setting Up the Module and Spy in Your Test Suite (once)
+- Include 'pytest_simcore.httpx_calls_capture' in your `pytest_plugins`.
+- Implement `create_httpx_async_client_spy_if_enabled("module.name.httpx.AsyncClient")` within your codebase.
+
+## Creating Mock Captures (every time you want to create/update the mock)
+- Initialize the real backend.
+- Execute tests using the command: `pytest --spy-httpx-calls-enabled=true --spy-httpx-calls-capture-path="my-captures.json"`.
+- Terminate the real backend once testing is complete.
```

Comment:
"once *capturing* is complete"
I guess this is all going on before the actual test is running.

Reply:
When the test finishes, all the captures are dumped.